Scripts/MCC22/competition-scripts/tapaal.sh

```diff
@@ -0,0 +1,540 @@
+
+    echo "$TMP"
+
+    SOLVED=$(echo "$TMP" | grep -oP "(?<=Query index )[0-9]+(?= was solved)")
+    SOLVED=$(echo $SOLVED | tr " " "\n" | sort -g -r)
+
+    for rem in $SOLVED ; do 
+            QUERIES=(${QUERIES[@]:0:$rem} ${QUERIES[@]:$(($rem + 1))})
+            echo "Solution found by multiquery processing (step 3) for query index" $rem 
+    done
+  fi
+
+
+    for trial in $(seq 0 20); do
+        time_left
+        if [[ ${#QUERIES[@]} == 0 ]]; then echo "All queries are solved" ; time_left; rm $QF; rm $MF; exit; fi
+        # Step 4: Parallel random search
+        echo "---------------------------------------------------"
+        echo "            Step 4: Random Parallel processing     "
+        echo "---------------------------------------------------"
+        echo "Doing random parallel verification of individual queries (${#QUERIES[@]}  in total)"
+        RUN_TIME=$(echo "$SECONDS/${#QUERIES[@]}" | bc)
+        echo "Each query is verified by ${#STRATEGIES_RAND[@]} parallel strategies for $RUN_TIME seconds"
+        
+        i=0 
+        for Q in ${QUERIES[@]}; do
+            echo "------------------- QUERY ${Q} ----------------------"
+            # Execute verifypn on all parallel strategies
+            # All processes are killed if one process provides an answer 
+            RUN_TIME=$(( $RUN_TIME < $SECONDS ? $RUN_TIME : $SECONDS))
+            if [[ "$RUN_TIME" -le 0 ]] ; then echo "Out of time, terminating!"; time_left; rm $QF; rm $MF; exit; fi
+            step1="$($PAR_CMD --line-buffer --halt now,success=1 --timeout $RUN_TIME --xapply\
+                eval $TIME_CMD $VERIFYPN -n $OPTIONS {} $MF $QF --binary-query-io 1 -x $Q -n \
+                ::: "${STRATEGIES_RAND[@]}" 2>&1)"
+
+            if [[ $? == 0 ]]; then
+                #unset QUERIES[$Q-1]
+                QUERIES=(${QUERIES[@]:0:$i} ${QUERIES[@]:$(($i + 1))})
+                i=$(echo "$i - 1" | bc)
+                echo "Solution found in random processing (step 4)"
+            else
+                echo "No solution found"
+            fi
+            FORMULA_RESULT="$(echo "$step1"|grep -m 1 FORMULA)"
+        
+            if [[ -n "$FORMULA_RESULT" ]]; then 
+                echo "$step1" | sed "/$FORMULA_RESULT/d"
+            else 
+                echo "$step1"
+            fi
+            echo 
+            echo "$FORMULA_RESULT"
+            time_left
+            i=$(echo "$i + 1" | bc)
+        done
+    # Exit if all queries are answered
+    done
+ 
+    time_left
+    echo "End of script."
+    rm $QF
+    rm $MF
+}
+
+function LTL {
+    unset STRATEGIES_PAR
+    STRATEGIES_PAR[0]="-ltl tarjan --ltl-heur dist -q 0 -l 0 -d $SHORTRED"
+    STRATEGIES_PAR[1]="-ltl tarjan --ltl-heur fire-count -q 0 -l 0 -d $SHORTRED"
+    STRATEGIES_PAR[2]="-ltl tarjan --ltl-heur aut -s DFS -p -q 0 -l 0 -d $SHORTRED"
```

Comment:
Heuristic setting is a no-op when using DFS search order, probably should be
```suggestion
    STRATEGIES_PAR[2]="-ltl tarjan --ltl-heur aut -p -q 0 -l 0 -d $SHORTRED"
```
for automaton heuristic without POR (sequential config is then with POR)

Reply:
`aut` is a parameter to `--ltl-heur`, it is listed in options when I compile the latest mainline (c2fcb4a0)